homeassistant/components/smartenergy_awattar/__init__.py

```diff
@@ -0,0 +1,215 @@
+
+
+def _setup_api(config: ConfigType) -> dict:
+    """Initialize the API and save the reference in the state."""
+    country: str = config[DOMAIN].get(CONF_COUNTRY)
+    _LOGGER.debug("Configuring Awattar API")
+    awattar_api = init_state(AWATTAR_API_URL[country])
+    _LOGGER.debug("Configured Awattar API")
+
+    return awattar_api
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up the Awattar defined via the UI.
+
+    - setup of the API
+    - coordinator
+    - sensors.
+    """
+    data = dict(config_entry.data)
+    entry_id: str = config_entry.entry_id
+
+    _LOGGER.debug(
+        "Setting up an entry with id=%s",
+        entry_id,
+    )
+
+    # scan interval is provided as an integer, but has to be an interval
+    scan_interval: timedelta = timedelta(seconds=data[CONF_SCAN_INTERVAL])
+    country: str = data[CONF_COUNTRY]
+
+    _LOGGER.debug("Configuring Awattar API")
+    hass.data[DOMAIN][INIT_STATE] = init_state(AWATTAR_API_URL[country])
+
+    # handle platform not ready
+    await ping_awattar(hass)
+
+    await _setup_coordinator(
+        hass,
+        scan_interval,
+        f"{entry_id}_coordinator",
+    )
+
+    hass.data[DOMAIN][entry_id] = data
+
+    for platform in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(config_entry, platform)
+        )
+
+    unsub_options_update_listener: Callable = config_entry.add_update_listener(
+        options_update_listener
+    )
+    hass.data[DOMAIN][INIT_STATE][UNSUB_OPTIONS_UPDATE_LISTENER][
+        entry_id
+    ] = unsub_options_update_listener
+
+    _LOGGER.debug("Entry setup for %s completed", entry_id)
+
+    return True
+
+
+async def options_update_listener(
+    hass: HomeAssistant, config_entry: ConfigEntry
+) -> None:
+    """Handle options update."""
+    await hass.config_entries.async_reload(config_entry.entry_id)
+
+
+async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    entry_id: str = config_entry.entry_id
+    _LOGGER.debug("Unloading the entry=%s", entry_id)
+
+    unloaded_platforms: list[tuple[list, str]] = [
+        (
+            await asyncio.gather(
+                *[
+                    hass.config_entries.async_forward_entry_unload(
+                        config_entry, platform
+                    )
+                ]
+            ),
+            platform,
+        )
+        for platform in PLATFORMS
+    ]
+    unload_ok: bool = all(unloaded_platforms)
+
+    # Remove options_update_listener.
+    hass.data[DOMAIN][INIT_STATE][UNSUB_OPTIONS_UPDATE_LISTENER][entry_id]()
+
+    # Remove config entry from the domain.
+    if unload_ok:
+        hass.data[DOMAIN][INIT_STATE] = {}
+
+    _LOGGER.debug("Unloaded the entry=%s", entry_id)
+
+    return unload_ok
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
```

Comment:
As your integration is communicating with a service/device it should only be setup via UI as specified in the [ADR](https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md).

Please remove this method and all yaml configuration related parts